File: Java/src/B7568.java
import java.util.Scanner;

public class B7568 {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		int n = sc.nextInt(); // n만큼 작동
		int[][] weight = new int [n][2]; // 덩치 저장
		int[] rank = new int [n];
		
		for(int i = 0; i < n; i++) { // 덩치 구하기
			int kg = sc.nextInt();
			int height = sc.nextInt();
			
			// input
			weight[i][0] = kg;
			weight[i][1] = height;
			rank[i] = 1; // 초기화
		}
		
		for(int i = 0; i < n; i++) { // 브루트포스 알고리즘
			
			for(int j = 0; j < n; j++) {
				
				if(i == j) { // 자기 자신과 비교 불가능
					continue;
				}
				// 키와 몸무게가 모두 커야 덩치가 큰 것임
				if(weight[i][0] < weight[j][0] && weight[i][1] < weight[j][1]) { // 덩치가 작으면 순위 밀림
					rank[i]++;
				}
			}
		}
		
		for(int i = 0; i < n; i++) {
			System.out.print(rank[i] + " ");
		}
		
		sc.close();
	}

}

File: 2021ing_python/B7568.py
N = int(input()) # 사람 수

people = [ # 사람 정보 리스트
    { 'weight': 0, 'height': 0, 'rank': 1} for _ in range(N)
]

# input
for i in range(N):
    w, h = map(int, input().split())

    people[i]['weight'] = w
    people[i]['height'] = h

# compare
for i in range(N):
    for j in range(N):
        if people[i]['weight'] < people[j]['weight'] and people[i]['height'] < people[j]['height']:
            people[i]['rank'] += 1

# print
for _ in range(N):
    print(people[_]['rank'], end=" ")
